### DISCUSSION TITLE
Query by publishDate is not working

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#702] [Q&A] [answered]
## Background

I want to have a simple scheduled publish feature like this

https://github.com/payloadcms/payload/discussions/567

## Repro Steps

1. start server
```
git clone git@github.com:Stupidism/seo-mono.git
git checkout feng/test-scheduled-publish
pnpm I
nx run seo-payload:serve
```

2. Create a doc and publish for the first time

http://localhost:3333/admin/collections/test-drafts/62be8d54722f5bd46633ca9f

<img width="1563" alt="image" src="https://user-images.githubusercontent.com/5801015/176833617-18414535-77cb-43f4-b6e2-3f844834acd5.png">

3. Query the doc with playground (to simulate guest access)

```
# Write your query or mutation here
query findTestDrafts {
    TestDrafts {
      docs {
        name
        publishDate
      }
    }
  }
```

<img width="1206" alt="image" src="https://user-images.githubusercontent.com/5801015/176833580-11f1398a-5d6e-4170-835c-606e672fde78.png">

4. Update the doc with a publishDate after today:
<img width="1666" alt="image" src="https://user-images.githubusercontent.com/5801015/176833741-f5641998-5f08-4fe5-937b-f6990d3c6179.png">

5. query the doc with playground again
Now u can see the updated version instead of the former version:

<img width="1335" alt="image" src="https://user-images.githubusercontent.com/5801015/176833825-fc31a3dc-d0ae-4108-8419-82c5ea439ee5.png">

### Expected behavior

According to requirement and the description from the code snnipet, it should return the version that publishDate smaller than today.

**Comment by DanRibbens** [comment, marked as answer]
Hey @Stupidism,

I'm glad to see you are exploring this solution more. Looking at the code for the access control, it is doing exactly what you're asking it to.
The access control on TestDrafts for `read` first checks if a user is logged in before querying publishDate. Unless you have logged out it won't bother querying the way you are expecting.

When you are querying in step 5, try from an incognito window or add a step to logout from the admin UI first.

I was able to do this from your repro to be sure.
![image](https://user-images.githubusercontent.com/6434612/176883183-369463cc-1ce0-4126-ada0-0adbc4d31d7a.png)

If your app will have authenticated users and admin users for the panel either by role or using a separate auth collection, remember to add those conditions to your read access function as well to gate the content.

**Reply by Stupidism** [reply]
Aha, I get what you mean now.
 
My understanding was wrong. I was expecting to see the `Before` version, when `After` version is not published

**Reply by DanRibbens** [reply]
Ah! Sure, you could still have that but you'll need to modify the query to use an `or` operator to allow the before to come through under the different conditions.
